code/network/multi_turret_manager.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "globalincs/pstypes.h"
+#include "globalincs/systemvars.h"
+#include "network/multi_time_manager.h"
+#include "network/multiutil.h"
+#include "ship/ship.h"
+
+typedef std::pair<ushort,int> turret_packet_address;
+
+struct turret_packet {
+    TIMESTAMP time;
+    int target_objnum;
+    std::pair<bool, float> angle1;
+    std::pair<bool, float> angle2;
+
+    turret_packet() = default;
+    
+    turret_packet(int time_in, ushort target_netsig, std::pair<bool, float> ang1_in, std::pair<bool, float> ang2_in) 
+    {
+        time = _timestamp(time_in + Multi_Timing_Info.get_mission_start_time());
+        target_objnum = OBJ_INDEX(multi_get_network_object(target_netsig));
```

Comment:
`multi_get_network_object()` can return `nullptr`, so you should always check its return value before use. Particularly in this case since it would result in an invalid `target_objnum`. You should get the target object, verify it's valid, then set `target_objnum` accordingly (OBJ_INDEX or -1).